From fec735ef2ee220900f3e82d8058a012bacb60ec7 Mon Sep 17 00:00:00 2001
From: Alyssa Rosenzweig <alyssa@rosenzweig.io>
Date: Tue, 14 Nov 2023 13:11:49 +0100
Subject: [PATCH 13/28] panfrost: Patch panfrost_max_thread_count() for v10

Not sure that's correct because some v10 GPUs have less 2k threads
allowed per core. We should probably use THREADS_MAX register value,
and only if it's zero, pick a default value that's the minimum of all
known v10 GPUs.

Signed-off-by: Alyssa Rosenzweig <alyssa@rosenzweig.io>
---
 src/panfrost/util/pan_ir.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/panfrost/util/pan_ir.h b/src/panfrost/util/pan_ir.h
index 5551fc7526a..aa96781fa74 100644
--- a/src/panfrost/util/pan_ir.h
+++ b/src/panfrost/util/pan_ir.h
@@ -452,6 +452,9 @@ panfrost_max_thread_count(unsigned arch, unsigned work_reg_count)
       return work_reg_count > 32 ? 384 : 768;
 
    /* Valhall (for completeness) */
+   case 10:
+      return 2048;
+
    default:
       return work_reg_count > 32 ? 512 : 1024;
    }
-- 
2.43.0

